tests/components/rainforest_raven/test_config_flow.py

```diff
@@ -0,0 +1,334 @@
+        yield comports
+
+
+async def test_flow_usb(hass: HomeAssistant, mock_comports, mock_device):
+    """Test usb flow connection."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={CONF_SOURCE: SOURCE_USB}, data=DISCOVERY_INFO
+    )
+    assert result
+    assert result.get("type") == data_entry_flow.FlowResultType.FORM
+    assert not result.get("errors")
+    assert result.get("flow_id")
+    assert result.get("step_id") == "meters"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input={CONF_MAC: METER_LIST.meter_mac_ids[0].hex()}
+    )
+    assert result
+    assert result.get("type") == data_entry_flow.FlowResultType.CREATE_ENTRY
+
+
+async def test_flow_usb_cannot_connect(
+    hass: HomeAssistant, mock_comports, mock_device_no_open
+):
+    """Test usb flow connection error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={CONF_SOURCE: SOURCE_USB}, data=DISCOVERY_INFO
+    )
+    assert result
+    assert result.get("type") == data_entry_flow.FlowResultType.ABORT
+    assert result.get("reason") == "cannot_connect"
+
+
+async def test_flow_usb_timeout_connect(
+    hass: HomeAssistant, mock_comports, mock_device_timeout
+):
+    """Test usb flow connection timeout."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={CONF_SOURCE: SOURCE_USB}, data=DISCOVERY_INFO
+    )
+    assert result
+    assert result.get("type") == data_entry_flow.FlowResultType.ABORT
+    assert result.get("reason") == "timeout_connect"
+
+
+async def test_flow_usb_parse_error(
+    hass: HomeAssistant, mock_comports, mock_device_parse_error
+):
+    """Test usb flow connection failure to parse."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={CONF_SOURCE: SOURCE_USB}, data=DISCOVERY_INFO
+    )
+    assert result
+    assert result.get("type") == data_entry_flow.FlowResultType.ABORT
+    assert result.get("reason") == "cannot_connect"
+
+
+async def test_flow_user(hass: HomeAssistant, mock_comports, mock_device):
+    """Test user flow connection."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={CONF_SOURCE: SOURCE_USER},
+    )
+    assert result
+    assert result.get("type") == data_entry_flow.FlowResultType.FORM
+    assert not result.get("errors")
+    assert result.get("flow_id")
+    assert result.get("step_id") == "user"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={
+            CONF_DEVICE: usb.human_readable_device_name(
+                DISCOVERY_INFO.device,
+                DISCOVERY_INFO.serial_number,
+                DISCOVERY_INFO.manufacturer,
+                DISCOVERY_INFO.description,
+                DISCOVERY_INFO.vid,
+                DISCOVERY_INFO.pid,
+            )
+        },
```

Comment:
This `user_input` is repeated identically many times in this test file; should it become a constant (like `DISCOVERY_INFO` already is) or a fixture..?

Reply:
Done in 329c19e.